### DISCUSSION TITLE
[Feature] Consider the date of birth when recognizing faces

### DISCUSSION THREAD

**Discussion by Germs2004** [immich-app/immich#12932] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

If I set a person's Date of Birth in Immich, the AI facial recognition should consider that information along with the photo's Date Photo Taken metadata when deciding which person that face belongs to. It should usually not link an adult's face in a photo to a person who was only 3 years old on the date the photo was taken.

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by bo0tzz** [comment]
#4743

**Reply by Germs2004** [reply]
It's not exactly a duplicate.  #4743 just asks to exclude people from being guessed in photos taken before they were born, but it wouldn't reduce the chance that a baby's face could be guessed as a 90 year old person or visa versa.

This new suggestion is to give the "Date of Birth" and "Date Photo Taken" to the AI so it can consider those two additional data points for all guesses of all ages.  The AI should be able to recognize what a baby, toddler, teen, adult, & senior face looks like, so having those two extra data points will help it make smarter guesses.

I think both of these requests would work well together, with #4743 just being a strict guardrail for the AI to prevent the most obvious mistakes.

Thank you for this project, by the way.  It's the closest in facial recognition quality to the old Google Picasa I've seen so far.  Better than Adobe and Digikam by far.

**Reply by bo0tzz** [reply]
>  The AI should be able to recognize what a baby, toddler, teen, adult, & senior face looks like

This has been requested in another FR before and it's not feasible - ML age recognition is simply not accurate enough to be useful.
